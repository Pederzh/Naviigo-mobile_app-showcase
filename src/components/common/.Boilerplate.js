import React from 'react';
import {StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import {Layout, withStyles} from "@ui-kitten/components";

const BoilerplateComponent = (props) => {
    const {themedStyle} = props;
    return(
        <React.Fragment>
            <Layout level='1'>

            </Layout>
        </React.Fragment>
    );
};

export const Boilerplate = withStyles(BoilerplateComponent, (theme) => {

    return ({


    });
});

const style = StyleSheet.create({

});