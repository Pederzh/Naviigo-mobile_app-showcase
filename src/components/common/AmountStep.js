import {useDispatch, useSelector} from "react-redux";
import {StyleSheet, View} from "react-native";
import {Button, Icon, Text} from "@ui-kitten/components";
import {countDecrement, countIncrement} from "../../actions/SearchActions";
import React from "react";

export const AmountStep = ({counterType}) => {
    const dispatch = useDispatch();
    //Access Redux Store State
    const people = useSelector((state) => state.people);

    const buttonEnabled = () => {
        return people[counterType] > 0;
    };

    return (
        <View style={style.amountContainer}>
            <Button
                style={[style.iconButton, style.amountButton]}
                size='small'
                appearance='outline' status='primary'
                icon={() => (<Icon {...style} name='minus'/>)}
                onPress={() => dispatch(countDecrement(counterType))}
                disabled={!buttonEnabled()}
            />
            <Text
                style={style.amount}
                category='s1'>
                {people[counterType]}
            </Text>
            <Button
                style={[style.iconButton, style.amountButton]}
                size='small'
                appearance='outline' status='primary'
                icon={() => (<Icon {...style} name='plus'/>)}
                onPress={() => dispatch(countIncrement(counterType))}
            />
        </View>
    );
};

const style = StyleSheet.create({
    container: {
        alignItems: 'flex-start',
        paddingHorizontal: 0,
        paddingVertical: 0,
    },
    detailsContainer: {
        flex: 1,
        height: '100%',
        padding: 16,
    },
    amountContainer: {
        flexDirection: 'row',
    },
    amountButton: {
        borderRadius: 16,
    },
    amount: {
        textAlign: 'center',
        top: 5,
        width: 40,
    },
    iconButton: {
        paddingHorizontal: 0,
    },
    itemContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        borderColor: "black"
    },
});