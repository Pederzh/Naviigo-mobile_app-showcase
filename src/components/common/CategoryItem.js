import React from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import {Button, withStyles} from "@ui-kitten/components";
import {TextualInfoItem} from "./TextualInfoItem";

const CategoryItemComponent = (props) => {
    const {themedStyle, children} = props;
    return(
        <View style={themedStyle.categoryContainer}>
        <Button
            style={themedStyle.categoryItem}
            status={props.status}
            size={props.size}
            appearance={props.appearance}
        >
            {children}
        </Button>
        </View>
    );
};

export const CategoryItem = withStyles(CategoryItemComponent, (theme) => {

    return ({
        categoryContainer: {
            flexDirection: 'row',
            flexWrap: 'wrap',
            marginHorizontal: -4,
        },
        categoryItem: {
            marginHorizontal: 4,
            borderRadius: 16,
        },
    });
});

const style = StyleSheet.create({

});

CategoryItem.defaultProps = {
    status: 'primary',
    size: 'tiny',
    appearance: 'filled',

};

CategoryItem.propTypes = {
    status: PropTypes.string,
    size: PropTypes.string,
    appearance: PropTypes.string,
};