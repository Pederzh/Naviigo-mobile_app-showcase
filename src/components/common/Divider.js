import React from 'react';
import {View} from 'react-native';

export const Divider = () => (
    <View
        style={{
            paddingVertical: 10,
            borderBottomColor: 'gainsboro',
            borderBottomWidth: 1,
        }}
    />
);