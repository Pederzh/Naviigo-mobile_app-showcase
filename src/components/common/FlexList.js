import React from 'react';
import {StyleSheet} from 'react-native';
import {List, withStyles} from "@ui-kitten/components";
import {useScreenInfo} from "../../hooks";

const FlexListComponent = (props) => {
    const {style, numColumns, contentContainerStyle, data, renderItem, ListHeaderComponent, ListFooterComponent, themedStyle} = props;
    const screenInfo = useScreenInfo();
    const {height, width, orientation, aspectRatio} = screenInfo;

    return (
        <List
            style={style}
            contentContainerStyle={contentContainerStyle}
            data={data}
            renderItem={renderItem}
            ListHeaderComponent={ListHeaderComponent}
            ListFooterComponent={ListFooterComponent}
            numColumns={numColumns ? props.numColumns : ((aspectRatio > 1.6) ? 1 : 2)}
            {...props}
        />
    );
};

export const FlexList = withStyles(FlexListComponent, (theme) => {
    return ({});
});
