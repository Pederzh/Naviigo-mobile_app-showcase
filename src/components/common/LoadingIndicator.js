import React from 'react';
import {StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import {Spinner, withStyles} from "@ui-kitten/components";

const LoadingIndicatorComponent = (props) => {
    const {themedStyle} = props;
    return(
        <View style={[themedStyle, themedStyle.indicator]}>
            <Spinner size='small'/>
        </View>
    )
};

export const LoadingIndicator = withStyles(LoadingIndicatorComponent, (theme) => {
    return ({
        indicator: {
            justifyContent: 'center',
            alignItems: 'center',
        },
    });
});