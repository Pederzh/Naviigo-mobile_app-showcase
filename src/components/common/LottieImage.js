import React from "react";
import {Platform} from "react-native";
import LottieView from "lottie-react-native";
import {Layout, withStyles} from '@ui-kitten/components';

const lottieImg = (type) => {
    switch (type) {
        case 'form-add-info':
            return (require('../../core/lottie/edit-pen.json'));
        case 'form-add-settings':
            return (require('../../core/lottie/tune-settings.json'));
        case 'form-set-map':
            return (require('../../core/lottie/set-map.json'));
        case 'form-add-image':
            return (require('../../core/lottie/add-image.json'));
        case 'form-end':
            return (require('../../core/lottie/clap-hands.json'));
        case 'empty-status':
            return (require('../../core/lottie/empty-status-1.json'));
        case 'success-status':
            return (require('../../core/lottie/success.json'));
        case 'error-status':
            return (require('../../core/lottie/error-dog.json'));
        default:
            return (null)
    }
};

const LottieImageComponent = ({themedStyle, type, loop, aspectRatio}) => (

    <Layout style={themedStyle.container}>
        {
            (Platform.OS === 'ios' && lottieImg(type) != null) ? (
                <LottieView
                    style={{
                        flexGrow: 1,
                        width: 120,
                        aspectRatio: (aspectRatio) ? aspectRatio : 1,
                        alignSelf: 'center',
                    }}
                    source={lottieImg(type)}
                    autoPlay
                    loop={loop != null ? loop : true}
                />
            ) : (
                <Layout style={themedStyle.spinnerContainer}>
                </Layout>
            )
        }

    </Layout>
);

export const LottieImage = withStyles(LottieImageComponent, (theme) => {
    return ({
        container: {
            flex: 1,
            height: 300,
            width: 300,
        },
        spinnerContainer: {
            flex: 1,
            marginTop: 240,
            justifyContent: 'center',
            alignItems: 'center'
        }
    });
});