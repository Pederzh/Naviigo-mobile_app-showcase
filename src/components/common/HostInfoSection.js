import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Avatar, Layout, Text, withStyles} from "@ui-kitten/components";
import {CategoryItem} from "./CategoryItem";
import {capitalizeFirstLetter, sanitizeUsername} from "../../utils";
import {useTranslation} from "react-i18next";

const HostInfoSectionComponent = (props) => {
    const { t } = useTranslation();
    const {themedStyle} = props;
    return (
        <React.Fragment>
            {
                (props?.position === 'right') ? (
                    <Layout level='1'>
                        <View style={themedStyle.commentHeaderRight}>
                            <View style={themedStyle.commentAuthorContainerRight}>
                                <Text category='s2'>
                                    {capitalizeFirstLetter(sanitizeUsername(props.name))}
                                </Text>
                                {
                                    (props.type) &&
                                    <CategoryItem
                                        appearance={'outline'}
                                    >
                                        {props.type}
                                    </CategoryItem>
                                }
                            </View>
                            <Avatar
                                style={{marginTop: 4}}
                                size='large'
                                source={{uri: `S3_URL/public/users/${sanitizeUsername(props?.name)}.jpg`}}
                            />
                        </View>
                    </Layout>
                ) : (
                    <Layout level='1'>
                        <View style={themedStyle.commentHeader}>
                            <Avatar
                                style={{marginTop: 4}}
                                size='large'
                                source={{uri: `S3_URL/public/users/${sanitizeUsername(props?.name)}.jpg`}}
                            />
                            <View style={themedStyle.commentAuthorContainer}>
                                <Text category='s2'>
                                    {capitalizeFirstLetter(sanitizeUsername(props.name))}
                                </Text>
                                {
                                    (props.type) &&
                                    <CategoryItem
                                        appearance={'outline'}
                                    >
                                        {t(props.type)}
                                    </CategoryItem>
                                }
                            </View>

                        </View>
                    </Layout>
                )
            }
        </React.Fragment>
    );
};

export const HostInfoSection = withStyles(HostInfoSectionComponent, (theme) => {

    return ({
        commentHeaderRight: {
            marginVertical: 8,
            justifyContent: 'flex-end',
            flexDirection: 'row',
        },
        commentAuthorContainerRight: {
            marginHorizontal: 16,
            alignItems: 'flex-end',
        },
        commentHeader: {
            marginVertical: 8,
            flexDirection: 'row',
        },
        commentAuthorContainer: {
            marginHorizontal: 16,
        },
    });
});

const style = StyleSheet.create({});