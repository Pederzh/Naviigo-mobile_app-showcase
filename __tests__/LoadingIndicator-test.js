import React from 'react';
import renderer from 'react-test-renderer';
import {LoadingIndicator} from '../src/components/common/LoadingIndicator';
import {
  ApplicationProvider
} from '@ui-kitten/components';
import {Provider} from 'react-redux';
import store from '../src/store/store';
import { mapping, light as lightTheme, dark as darkTheme } from '@eva-design/eva';

// define the props that should be passed to the component
const boat = { item: {name: 'Boat 1', image: require('naviigo/assets/img/boat1.jpg')}}
const key = 1;
//const value = []
const onchange = () => {};
const selectedoption = 1;
const day = {};
const multiple = true;

const data = {
  find: () => {}
};

test('renders correctly', () => {
  const tree = renderer.create(   
    <Provider store={store}>
          <ApplicationProvider mapping={mapping} theme={lightTheme}>
            <LoadingIndicator />
          </ApplicationProvider>
        </Provider>
          ).toJSON();
  expect(tree).toMatchSnapshot();
});