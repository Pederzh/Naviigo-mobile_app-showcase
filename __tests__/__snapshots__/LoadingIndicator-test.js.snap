// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "indicator": Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Size {
          "height": 20,
          "width": 20,
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "height": 20,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 20,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              undefined,
              Size {
                "height": 10,
                "width": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "height": 20,
                "transform": Array [
                  Object {
                    "rotate": "-165deg",
                  },
                ],
                "width": 20,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Size {
                    "height": 10,
                    "width": 20,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#3366FF",
                      "borderRadius": 10,
                      "borderWidth": 2.8,
                      "height": 20,
                      "width": 20,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "bottom": 0,
            "height": 20,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 20,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "top": 10,
              },
              Size {
                "height": 10,
                "width": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "height": 20,
                "transform": Array [
                  Object {
                    "translateY": -10,
                  },
                  Object {
                    "rotate": "345deg",
                  },
                ],
                "width": 20,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Size {
                    "height": 10,
                    "width": 20,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#3366FF",
                      "borderRadius": 10,
                      "borderWidth": 2.8,
                      "height": 20,
                      "width": 20,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
